### ISSUE TITLE
Unable to change schema for migrations

### ISSUE THREAD

**Issue by ikaoullas** [flyway/flyway#1995]
###### Which version and edition of Flyway are you using?
I am using v5.0.7 . I am trying to upgrade from version v4.0.3 where I was able to change the schema for the migrations

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
Java API

###### Which database are you using (type & version)?
H2

###### Which operating system are you using?
Ubuntu 16.04

###### What did you do?
(Please include the content causing the issue, any relevant configuration settings, the SQL statement that failed (if relevant) and the command you ran.)
I have some tests running that use a different schema than the default one. We use a properties file with the following properties:
```
h2.db.user=h2
h2.db.password=h2
db.schema.name=my_schema
h2.db.driver.class=org.h2.Driver
h2.db.dialect=org.hibernate.dialect.H2Dialect
h2.jpa.database=H2
h2.db.url=jdbc:h2:mem:${db.schema.name};AUTOCOMMIT=OFF;MVCC=TRUE;DB_CLOSE_DELAY=-1;DATABASE_TO_UPPER=false
h2.hibernate.ddl-auto=none
```
During the test setup phase we are using the following code to migrate the database to the target version:
```
Flyway flyway = new Flyway();
flyway.setDataSource(dataSource);
flyway.setLocations(env.getProperty("spring.flyway.locations"));
flyway.setTarget(targetVersion);

if (schemas != null) {
        flyway.setSchemas(schemas);
}
flyway.setCallbacks(callbacks);

try {
     flyway.migrate();
} catch (Exception e) {
     logger.error("Migration Failed", e);
}
```

###### What did you expect to see?
Before the upgrade to v5.0.7 I was able to run all the migrations and make changes to the tables in the schema provided in our properties file. I was expecting to see the same behaviour after the upgrade.

###### What did you see instead?
After the upgrade to v5.0.7 our Java migrations were failing because they were trying to access tables under the schema we provided in our properties file but the tables were missing. 
By running the query `SELECT TABLE_NAME, TABLE_SCHEMA FROM INFORMATION_SCHEMA.TABLES` I could see that all the tables created in the previous migrations before my first Java migration were created under the PUBLIC schema which is the default one for H2.

I noticed that in the H2Database instance, the main connection object had the correct schema set, but during the migrations the getMigrationConnection() method was being called and it was always using the default schema name..

Please let me know if you need more information

**Comment by axelfontaine** [comment]
I can't seem to reproduce this. Can you share a small repo that reproduces the issue?

**Comment by netmikey** [comment]
I ran into this as well. Our setup is as follows:

- Flyway 5.0.7
- PostgreSQL (latest version)
- Schema-path on connection: `"myuser_SomeSchema,public"` (Postgres default behavior afaik)
- User: `myuser`
- Schema we want to target: `myuser_SomeSchema`
- SQL migration scripts don't specify any schema

I'm currently seeing the issue while using the Gradle Task, but from what I've seen on our CI server, I suspect our actual Application's migration (which runs within Spring Boot) to be affected as well.

Here's the task configuration:
```
flywayMigrate {
    url = 'jdbc:postgresql://pgserver:5432/db?currentSchema=myuser_SomeSchema,public'
    user = 'myuser'
    password = 'mypassword'
    locations = [ 'filesystem:<some path...>' ]
    schemas = [ 'myuser_SomeSchema' ]
}
```

Whether I specify `schemas` or not yields the same result:
- Flyway creates the `flyway_schema_history` table in the `myuser_SomeSchema` schema as expected.
- Flyway seems to run the SQL migrations against the `public` schema, because all tables except the `flyway_schema_history` end up over there.

I checked with multiple Flyway versions: When I revert to 4.2.0, the issue is gone. It starts happening with 5.0.0.

I don't know whether this is related, but it seemed to have started happening only when we started using mixed-case schema names (like `user_SomeSchema` instead of `user_someschema`).
